specs/ExtendedProcessFailedEventArgs.md

```diff
@@ -0,0 +1,98 @@
+
+Code integrity failure source module path
+===
+
+# Background
+[Windows Code Integrity](https://learn.microsoft.com/en-us/mem/intune/user-help/you-need-to-enable-code-integrity) is a feature that verifies the
+integrity of the code that runs on the system. It helps protect it from malware,
+tampering, and unauthorized changes. Code integrity checks the digital
+signatures of the files that are loaded into memory, and prevents any
+file that does not have a valid signature from running in WebView2 process.
+We are extending ProcessFailedEventArgs with FailureSourceModulePath property
+which caused webview2 process to exit with code STATUS_INVALID_IMAGE_HASH.
+
+# Examples
+
+    ```c#
+        /// This is an event handler for our CoreWebView2's ProcessFailedEvent
+        private void CoreWebView2_ProcessFailed(object sender, CoreWebView2ProcessFailedEventArgs e)
+        {
+            if (e.ExitCode == -1073740760 /*STATUS_INVALID_IMAGE_HASH*/)
+            {
+                // If the process crashed because of STATUS_INVALID_IMAGE_HASH,
+                // then we want to log to our app's telemetry the name of the
+                // DLL that caused the issue.
+                SendTelemetry(e.FailureSourceModulePath);
+            }
+        }
+    ```
+
+    ```cpp
+    CHECK_FAILURE(m_webView->add_ProcessFailed(
+        Callback<ICoreWebView2ProcessFailedEventHandler>(
+            [this](ICoreWebView2* sender, ICoreWebView2ProcessFailedEventArgs* argsRaw)
+                -> HRESULT {
+                wil::com_ptr<ICoreWebView2ProcessFailedEventArgs> args = argsRaw;
+                int exit_code;
+                CHECK_FAILURE(args->get_ExitCode(&exit_code));
+
+                if (exit_code == -1073740760 /*STATUS_INVALID_IMAGE_HASH*/) {
```

Comment:
FailureSourceModulePath doesn't appear to be specific to STATUS_INVALID_IMAGE_HASH or Windows Code Integrity.  Is the expectation that this string could be populated in other failure cases, but STATUS_INVALID_IMAGE_HASH is the only case when we're currently promising it will always be populated?

Reply:
Yes that is exactly what I was thinking.